### ISSUE TITLE
TypeError: must start with buffer, array, number or string

### ISSUE THREAD

**Issue by mDomingues** [webdriverio/wdio-allure-reporter#7]
Guys, I've spoken with you in gitter about this bug I'm dealing with.
My test suites run with some faults to the end as expected but the Allure crashes with a message like:

![image](https://cloud.githubusercontent.com/assets/11029958/14256798/92a871b6-fa92-11e5-95bb-0d47de7de44a.png)

This happens when I use this lib.
[webdriverajax](https://www.npmjs.com/package/webdriverajax)

My code that causes crash is similar to this one, but the assertion fails!
```
browser
  .url('http://foo.bar')
  .setupInterceptor() // capture ajax calls 
  .expectRequest('GET', '/api/foo', 200) // expect GET request to /api/foo with 200 statusCode 
  .click('#button') // button that initiates ajax request 
  .pause(1000) // maybe wait a bit until request is finished 
  .assertRequests(); // validate the requests 
```

Hope you discover what's wrong :)

**Comment by just-boris** [comment]
@sullenor You have got something like this, maybe you have something to add.

**Comment by sullenor** [comment]
@just-boris Yeah, I had the same issue with `webdriver` and `wdio-allure-reporter`. Recently, I started to get this exception more often. 

I'd try to make a small example to reproduce this issue, may be it will be easier to fix it.

**Comment by sullenor** [comment]
Looks like I get this error in case allure can't find the file with screenshot. Despite that screenshot was successfully generated. Presumably, it can't resolve path to file.

**Comment by sullenor** [comment]
I tried to log filename here: https://github.com/webdriverio/webdriverio/blob/master/lib/runner.js#L91 and looks like it generates different paths: absolute in one case and only filename in another.

```bash
ERROR_firefox_2016-04-05T14-44-08.997Z.png
Trace
    at EventEmitter.<anonymous> (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/runner.js:121:41)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)
    at Object.<anonymous> (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/webdriverio.js:650:55)
    at Object.exec (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/helpers/safeExecute.js:28:24)
    at Object.resolve (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/webdriverio.js:171:29)
    at /home/sullenor/yharnam/node_modules/webdriverio/build/lib/webdriverio.js:384:32
    at Promise.apply (/home/sullenor/yharnam/node_modules/webdriverio/node_modules/q/q.js:1151:26)
    at Promise.promise.promiseDispatch (/home/sullenor/yharnam/node_modules/webdriverio/node_modules/q/q.js:774:41)
    at /home/sullenor/yharnam/node_modules/webdriverio/node_modules/q/q.js:1377:14
```

```bash
/home/sullenor/yharnam/errorShots/ERROR_firefox_2016-04-05T14-44-08.997Z.png
Trace
    at EventEmitter.<anonymous> (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/runner.js:121:41)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)
    at Object.<anonymous> (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/webdriverio.js:650:55)
    at Object.exec (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/helpers/safeExecute.js:28:24)
    at Object.resolve (/home/sullenor/yharnam/node_modules/webdriverio/build/lib/webdriverio.js:171:29)
    at /home/sullenor/yharnam/node_modules/webdriverio/build/lib/webdriverio.js:384:32
    at Promise.apply (/home/sullenor/yharnam/node_modules/webdriverio/node_modules/q/q.js:1151:26)
    at Promise.promise.promiseDispatch (/home/sullenor/yharnam/node_modules/webdriverio/node_modules/q/q.js:774:41)
    at /home/sullenor/yharnam/node_modules/webdriverio/node_modules/q/q.js:1377:14
```

**Comment by sullenor** [comment]
Presumably, the problem lies somewhere here: https://github.com/webdriverio/webdriverio/blob/master/lib/webdriverio.js#L218

Reporter should fetch the content of file before attaching it: https://github.com/webdriverio/wdio-allure-reporter/blob/master/lib/reporter.js#L162

**Comment by eben0** [comment]
I also face this issue a lot. When an element does not exist the payload becomes empty:

```
Failed parsing stats for Allure reporter: must start with number, buffer, array or string. Dumped this.baseReporter.stats to C:\Users\adm
TypeError: must start with number, buffer, array or string
    at fromObject (buffer.js:140:11)
    at new Buffer (buffer.js:62:10)
    at C:\Users\admin\AppData\Roaming\npm\node_modules\wdio-allure-reporter\build\reporter.js:286:131
    at Array.map (native)
    at AllureReporter.addOutput (C:\Users\admin\AppData\Roaming\npm\node_modules\wdio-allure-reporter\build\reporter.js:257:20)
    at AllureReporter.parseStats (C:\Users\admin\AppData\Roaming\npm\node_modules\wdio-allure-reporter\build\reporter.js:153:50)
    at AllureReporter.<anonymous> (C:\Users\admin\AppData\Roaming\npm\node_modules\wdio-allure-reporter\build\reporter.js:68:12)
    at emitOne (events.js:90:13)
    at AllureReporter.emit (events.js:182:7)
    at BaseReporter.handleEvent (C:\Users\admin\AppData\Roaming\npm\node_modules\webdriverio\build\lib\utils\BaseReporter.js:302:35)
```

if event.payload.data is empty, a quick dirty fix is to check and assign an empty string:

```
new Buffer(event.payload.data || '', 'base64')
```

**Comment by christian-bromann** [comment]
@eben0 @mDomingues can you please retry with the recent published version?

**Comment by mDomingues** [comment]
So far it seems fixed in version 1.4.23.
Good job!

**Comment by christian-bromann** [comment]
Cool! Closing then
